messaging/src/main/java/org/axonframework/eventhandling/deadletter/DeadLetteredEventProcessingTask.java

```diff
@@ -85,6 +86,8 @@ public EnqueueDecision<EventMessage<?>> process(DeadLetter<? extends EventMessag
         UnitOfWork<? extends EventMessage<?>> unitOfWork = DefaultUnitOfWork.startAndGet(letter.message());
 
         unitOfWork.attachTransaction(transactionManager);
+        unitOfWork.resources()
+                  .put(DeadLetterParameterResolverFactory.CURRENT_DEAD_LETTER, letter);
```

Comment:
I have seen a convention (I don't recall where), we resources of which you expect only a single one of its type are recorded using their class as the key of the resource. They had a shortcut `T get(Class<T> type)` on the resources that allows you to get these resources without requiring any casting.
While we don't necessarily use this convention, it should not be an issue to use `DeadLetter.class` as the key of the resource here.

Reply:
A fair and simple suggestion. I'll use the `DeadLetter.class#getName` as the key instead. Although it would be nice if the `UnitOfWork#resources` allowed insertion using a `Class` as key, I think that adjustment is a little out of scope for this PR.